# Assignment 3

### Instructions
1. Import the following modules:
    - `pandas`
    - `stats` from `scipy`
2. Create a dataframe from the data contained in *income_data.csv* using `pandas`

In [ ]:
import pandas as pd
from scipy import stats

df = pd.read_csv('./income_data.csv')
df.head()

,job,type,income,education,prestige
0,accountant,prof,62,86,82
1,pilot,prof,72,76,83
2,architect,prof,75,92,90
3,author,prof,55,90,76
4,chemist,prof,64,86,90


3. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *education*

In [2]:
# Income and Education
print('Income and Education')
r, p = stats.pearsonr(df['income'], df['education'])
print(f'r (Pearson correlation coefficient): {r:.3f}')
print(f'p-value:                             {p:.3f}')

print('')

Income and Education
r (Pearson correlation coefficient): 0.725
p-value:                             0.000



4. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *income* and *prestige*

In [3]:
# Income and Prestige
print('Income and Prestige')
r, p = stats.pearsonr(df['income'], df['prestige'])
print(f'r (Pearson correlation coefficient): {r:.3f}')
print(f'p-value:                             {p:.3f}')

print('')

Income and Prestige
r (Pearson correlation coefficient): 0.838
p-value:                             0.000



5. Calculate and display Pearson's **correlation coefficients** and the **p-values** for *education* and *prestige*

In [4]:
# Education and Prestige
print('Education and Prestige')
r, p = stats.pearsonr(df['education'], df['prestige'])
print(f'r (Pearson correlation coefficient): {r:.3f}')
print(f'p-value:                             {p:.3f}')

print('')

Education and Prestige
r (Pearson correlation coefficient): 0.852
p-value:                             0.000



6. In the empty `Markdown` cell below, explain the following for **each of the calculated correlation coefficients**
    - The **strength** and **direction** of the correlation
    - The **statistical significance** of the test (**p-value**); emphasize if we can trust the correlation test or not, and why

1. Income and Education

Correlation: The corelation is 0.725, indicating a high positive correlation.

Statistical significance: The p-value is near zero, so there is high statistical significance and we can trust the correlation.

2. Income and Prestige

Correlation: The corelation is 0.838, indicating a high positive correlation.

Statistical significance: The p-value is near zero, so there is high statistical significance and we can trust the correlation.

3. Education and Prestige

Correlation: The corelation is 0.852, indicating a high positive correlation.

Statistical significance: The p-value is near zero, so there is high statistical significance and we can trust the correlation.


In the empty `Markdown` cell below, answer the following question:

7. Pearson's correlation coefficient OR covariance can both be used to measure the direction of the relationship between two variables (e.g., education and income). Explain why Pearson's correlation coefficient may be better than covariance.

Person's correlation coefficent may be better than covariance because correlation is standardized (doesn't depend on the units of the two variable), while covariance depends on the units of measurement. For example, a correlation of 0.825 clearly indicates a strong positive relationship, regardless of whether education is measured in years and income in dollars.

8. Submit this `Jupyter` file to D2L, renamed as **Last_First_Assignment3.ipynb** 
    - Replace '**Last**' and '**First**' with your name